### PR TITLE
Implement context menu for record nav (fix #65)

### LINKED ISSUE #65: Support navigation menu
Support navigation menu where users can jump to any part of document easily.

### PR DESCRIPTION
**Pull request by harunurhan**
* Implement tree-menu and tree-menu-item components.

* Implement window-href service that gets window.location.href.

* Set value path as an id to each component in order to navigate via #.

* Introduce shared dom-utils service with focus, select and flash utilities.

* Focus the first input child when navigated to a component.

* Implement shallow search for menu and navigation by typing.

##### You can search things, and navigate to them by `enter` if you give the exact path.
type-tree-menu

##### Clicking things works too.
click-tree-menu

### CODE REVIEW

File: src/app/editor/editor.component.scss [hunk 1]
```diff
@@ -104,4 +104,8 @@ div.right-side-container {
   height: 100vh;
   padding-left: $container-padding;
   border-left: 2px solid lightgray;
+}
+
+.flash {
+  border: 2px solid yellow;
 }
```
Comment (kaplun) [inline src/app/editor/editor.component.scss]: I think you should configure your editor to add newlines at the end of files :)

Comment (harunurhan) [inline src/app/editor/editor.component.scss]: It will be in another pr where I fix tslint errors (which also forces to add new-line at the end of `*.ts` files.)
